### DISCUSSION TITLE
Image Becomes Blurry When Resized Using Corner Scaling in Fabric.js 6.4.2 with Angular.

### DISCUSSION THREAD

**Discussion by Arkit-Sutariya** [fabricjs/fabric.js#10323] [Q&A]
I am working on an Angular project and using Fabric.js version 6.4.2. I am adding an image from our server to the Fabric.js canvas. The image displays perfectly and looks clean initially. However, when I **resize** the image using the corner scaling controls to make it **smaller**, it becomes **blurry**, and the borders exhibit a **zig-zag visual effect**.

I have created a minimal reproducible example on CodeSandbox, which you can access [here](https://codesandbox.io/p/sandbox/sharp-https-xcvqd8).

Question:
    1. Is this issue related to Fabric.js's image rendering or scaling functionality?
    2. Could this be due to the quality of the server image itself?
    3. What are the best practices or recommended settings in Fabric.js to avoid such issues during scaling?

I would appreciate guidance on whether this is a Fabric.js-specific issue or something related to the source image, along with suggestions to resolve the problem.

Thank you!

Here is default load image.
![image](https://github.com/user-attachments/assets/69465816-c26b-489f-8888-ee4513fe6976)

After size from corner
![image](https://github.com/user-attachments/assets/fd688b20-b539-4467-a83e-fba91b5868ee)

**Comment by asturur** [comment]
Canvas uses nearest neighbour scaling for images and when a strong scaling factor is applied it looks like that.
If you need to make images very small you can look at using filters or the resizeFilter property that allows you to use our internal Resize filter.

Try to look at this demo for example
https://fabricjs.com/demos/lanczos-webgl/

(ooofff is currently broken going to fix the demo shortly)

**Reply by Arkit-Sutariya** [reply]
Thank you for your quick response.
I will definitely try this solution.

Could you please review below issue also  ?
https://github.com/fabricjs/fabric.js/discussions/10322

**Reply by asturur** [reply]
i have seen that, understanding what you are doing wrong with cors isn't easy. FabricJS does support cors.

**Reply by asturur** [reply]
https://fabricjs.com/demos/lanczos-webgl/

I think i fixed the demo

**Reply by asturur** [reply]
Maybe the demo needs to also speak specifically of your issue

**Reply by Arkit-Sutariya** [reply]
> i have seen that, understanding what you are doing wrong with cors isn't easy. FabricJS does support cors.

Thank you,
If you got any solution please share with me.

**Reply by Arkit-Sutariya** [reply]
> https://fabricjs.com/demos/lanczos-webgl/
> 
> I think i fixed the demo

I have found the solution.

**Reply by vaynevayne** [reply]
`transform  [0.17609841827768014, 0, 0, 0.17609841827768014, 0, 142.2884007029877]`
When the zoom factor is small, the rectangle will blur
